AlphaWallet/KeyManagement/EtherKeystore.swift

```diff
@@ -0,0 +1,647 @@
+        switch wallet.type {
+        case .real(let account):
+            return ethereumAddressesWithSeed.contains(account.address.eip55String)
+        case .watch:
+            return false
+        }
+    }
+
+    func isKeystore(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real(let account):
+            return ethereumAddressesWithPrivateKeys.contains(account.address.eip55String)
+        case .watch:
+            return false
+        }
+    }
+
+    func isWatched(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real:
+            return false
+        case .watch(let address):
+            return watchAddresses.contains(address.eip55String)
+        }
+    }
+
+    func isProtectedByUserPresence(account: EthereumAccount) -> Bool {
+        return ethereumAddressesProtectedByUserPresence.contains(account.address.eip55String)
+    }
+
+    func signPersonalMessage(_ message: Data, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        let prefix = "\u{19}Ethereum Signed Message:\n\(message.count)".data(using: .utf8)!
+        return signMessage(prefix + message, for: account)
+    }
+
+    func signHash(_ hash: Data, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        guard let key = getPrivateKeyForSigning(forAccount: account) else { return .failure(.accountMayNeedImportingAgainOrEnablePasscode) }
+        do {
+            var data = try EthereumSigner().sign(hash: hash, withPrivateKey: key)
+            // TODO: Make it configurable, instead of overriding last byte.
+            data[64] += 27
+            return .success(data)
+        } catch {
+            return .failure(KeystoreError.failedToSignMessage)
+        }
+    }
+
+    func signTypedMessage(_ datas: [EthTypedData], for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        let schemas = datas.map { $0.schemaData }.reduce(Data(), { $0 + $1 }).sha3(.keccak256)
+        let values = datas.map { $0.typedData }.reduce(Data(), { $0 + $1 }).sha3(.keccak256)
+        let combined = (schemas + values).sha3(.keccak256)
+        return signHash(combined, for: account)
+    }
+
+    func signMessage(_ message: Data, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        return signHash(message.sha3(.keccak256), for: account)
+    }
+
+    func signMessageBulk(_ data: [Data], for account: EthereumAccount) -> Result<[Data], KeystoreError> {
+        guard let key = getPrivateKeyForSigning(forAccount: account) else { return .failure(.accountMayNeedImportingAgainOrEnablePasscode) }
+
+        do {
+            var messageHashes = [Data]()
+            for i in 0...data.count - 1 {
+                let hash = data[i].sha3(.keccak256)
+                messageHashes.append(hash)
+            }
+            var data = try EthereumSigner().signHashes(messageHashes, withPrivateKey: key)
+            // TODO: Make it configurable, instead of overriding last byte.
```

Comment:
Fix TODO.

Reply:
It's a leftover from old code before the PR. I think we'll always have this hardcoded for Ethereum signatures. I'll fix it outside of the PR.